src/Uno.UI/Microsoft/UI/Xaml/Controls/ProgressBar/ProgressBar.cs

```diff
@@ -0,0 +1,249 @@
+			RegisterPropertyChangedCallback(MaximumProperty, OnRangeBasePropertyChanged);
+			RegisterPropertyChangedCallback(PaddingProperty, OnRangeBasePropertyChanged);
+
+			TemplateSettings = new ProgressBarTemplateSettings();
+		}
+
+		protected override AutomationPeer OnCreateAutomationPeer() => new ProgressBarAutomationPeer(this);
+
+		protected override void OnApplyTemplate()
+		{
+			m_layoutRoot = GetTemplateChild("LayoutRoot") as Grid;
+
+			// NOTE: Example of how named parts are loaded from the template. Important to remember that it's possible for
+			// any of them not to be found, since devs can replace the template with their own.
+
+			m_determinateProgressBarIndicator = GetTemplateChild("DeterminateProgressBarIndicator") as Rectangle;
+			m_indeterminateProgressBarIndicator = GetTemplateChild("IndeterminateProgressBarIndicator") as Rectangle;
+			m_indeterminateProgressBarIndicator2 = GetTemplateChild("IndeterminateProgressBarIndicator2") as Rectangle;
+
+			UpdateStates();
+		}
+
+		private void OnSizeChange()
+		{
+			SetProgressBarIndicatorWidth();
+			UpdateWidthBasedTemplateSettings();
+		}
+
+		private void OnRangeBasePropertyChanged(DependencyObject sender, DependencyProperty dp)
+		{
+			// NOTE: This hits when the Value property changes, because we called RegisterPropertyChangedCallback.
+			SetProgressBarIndicatorWidth();
+		}
+
+		private static void OnIsIndeterminateChanged(DependencyObject dependencyobject, DependencyPropertyChangedEventArgs args)
+		{
+			// NOTE: This hits when IsIndeterminate changes because we set MUX_PROPERTY_CHANGED_CALLBACK to true in the idl.
+			if (dependencyobject is ProgressBar progressBar)
+			{
+				progressBar.SetProgressBarIndicatorWidth();
+				progressBar.UpdateStates();
+			}
+		}
+
+		private static void OnShowErrorChanged(DependencyObject dependencyobject, DependencyPropertyChangedEventArgs args)
+		{
+			if (dependencyobject is ProgressBar progressBar)
+			{
+				progressBar.UpdateStates();
+			}
+		}
+
+		private static void OnShowPausedChanged(DependencyObject dependencyobject, DependencyPropertyChangedEventArgs args)
+		{
+			if (dependencyobject is ProgressBar progressBar)
+			{
+				progressBar.UpdateStates();
+			}
+		}
+
+		private void UpdateStates()
+		{
+			var showError = ShowError;
+			var isIndeterminate = IsIndeterminate;
+
+			if (showError && isIndeterminate)
+			{
+				VisualStateManager.GoToState(this, s_IndeterminateErrorStateName, true);
+			}
+			else if (showError)
+			{
+				VisualStateManager.GoToState(this, s_ErrorStateName, true);
+			}
+			else if (isIndeterminate && ShowPaused)
+			{
+				VisualStateManager.GoToState(this, s_IndeterminatePausedStateName, true);
+			}
+			else if (ShowPaused)
+			{
+				VisualStateManager.GoToState(this, s_PausedStateName, true);
+			}
+			else if (isIndeterminate)
+			{
+				UpdateWidthBasedTemplateSettings();
+				VisualStateManager.GoToState(this, s_IndeterminateStateName, true);
+			}
+			else if (!isIndeterminate)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change this condition so that it does not always evaluate to 'true'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5444000)